src/nsrt_learning.py

```diff
@@ -19,14 +18,40 @@ def learn_nsrts_from_data(dataset: Dataset, predicates: Set[Predicate],
     """
     print(f"\nLearning NSRTs on {len(dataset)} trajectories...")
 
-    transitions_by_option = generate_transitions(dataset, predicates)
+    # Segment transitions based on changes in predicates.
+    segments = [seg for traj in dataset
+                for seg in segment_trajectory(traj, predicates)]
 
+    # Learn strips operators.
+    strips_ops, partitions = learn_strips_operators(segments)
+    assert len(strips_ops) == len(partitions)
+
+    # Learn options, or if known, just look them up.
+    # The order of the options corresponds to the strips_ops.
+    # Each item is a (ParameterizedOption, Sequence[Variable])
+    # with the latter holding the option_vars.
+    option_specs = learn_options(strips_ops, partitions)
```

Comment:
In your learning idea, are you clear how the option_vars would fall out? They seem to be tied to the NSRT right?